src/jrd/replication/ChangeLog.cpp

```diff
@@ -893,13 +941,17 @@ void ChangeLog::clearSegments()
 ChangeLog::Segment* ChangeLog::createSegment()
 {
 	const auto state = m_sharedMemory->getHeader();
-	const auto sequence = ++state->sequence;
+	const auto sequence = state->sequence + 1;
 
 	PathName filename;
 	filename.printf(FILENAME_PATTERN, m_config->filePrefix.c_str(), sequence);
 	filename = m_config->journalDirectory + filename;
 
 	const auto fd = os_utils::openCreateSharedFile(filename.c_str(), O_EXCL | O_BINARY);
+	if (fd == -1)
+	{
+		raiseError("Journal file %s create() failed (error %d)", filename.c_str(), ERRNO);
+	}
```

Comment:
openCreateSharedFile() throws itself, so this addition is not necessary.

Reply:
Done, thanks